rust/connlib/snownet/src/allocation.rs

```diff
@@ -1470,6 +1471,30 @@ mod tests {
         )
     }
 
+    #[test]
+    fn allocation_refresh_timeout_is_reset_after_refresh_request_is_sent() {
+        let mut allocation = Allocation::for_test(Instant::now());
+
+        let allocate = allocation.next_message().unwrap();
+        allocation.handle_test_input(
+            &allocate_response(&allocate, &[RELAY_ADDR_IP4, RELAY_ADDR_IP6]),
+            Instant::now(),
+        );
+
+        let next_timeout = allocation.poll_timeout().unwrap();
+        assert_eq!(next_timeout, allocation.refresh_allocation_at().unwrap());
```

Comment:
I'd rsther we assert the actual time here by including a lifetime in the allocation response. I don't want to rely on the code under test for the assertion.